frontend/src/components/pages/RegistrantExperience/PersonalInfo/personalInfoReducer.tsx

```diff
@@ -0,0 +1,166 @@
+import React from "react";
+import {
+  AddCamper,
+  DeleteCamper,
+  PersonalInfoActions,
+  PersonalInfoReducerDispatch,
+  UpdateCamper,
+  UpdateContact,
+} from "../../../../types/PersonalInfoTypes";
+import { RegistrantExperienceCamper } from "../../../../types/CamperTypes";
+
+export const CamperReducer = (
+  setCampers: React.Dispatch<
+    React.SetStateAction<RegistrantExperienceCamper[]>
+  >,
+  action: PersonalInfoReducerDispatch,
+): void => {
+  setCampers((campers: RegistrantExperienceCamper[]) => {
+    const newCampers: RegistrantExperienceCamper[] = JSON.parse(
+      JSON.stringify(campers),
+    ); // Deep Copy
+    switch (action.type) {
+      case PersonalInfoActions.ADD_CAMPER: {
+        const { campSessions } = action as AddCamper;
+        // Add a unique camper entry for all registered camp sessions
+        /* eslint-disable-next-line */
+        const campSessionIds = Array.from(campSessions.map((campSession) => campSession.id));
+        newCampers.push({
+          id: "",
+          campSessions: campSessionIds,
```

Comment:
could you not map directly on the campSessions property like this: `campSessions: campSessions.map(...)` or does typescript complain about map returning void[]?

Reply:
Oh, this was just done for clarity on my end. Let me change it to use `campSessions: campSessions.map(...)`